tests/ui/EditTask.test.Edit_Modal_HTML_snapshot_tests_should_match_snapshot.approved.html

```diff
@@ -1,226 +1,222 @@
+      <code class="results">
+        📅
+        <i>no due date</i>
+      </code>
+      <label for="scheduled" class="accesskey-first">Scheduled</label>
+      <input id="scheduled" type="text" class="input" placeholder="Try 'Mon' or 'tm' then space" accesskey="s" />
+      <code class="results">
+        ⏳
+        <i>no scheduled date</i>
+      </code>
+      <label for="start">
+        St
+        <span class="accesskey">a</span>
+        rt
+      </label>
+      <input id="start" type="text" class="input" placeholder="Try 'Mon' or 'tm' then space" accesskey="a" />
+      <code class="results">
+        🛫
+        <i>no start date</i>
+      </code>
+      <div class="future-dates-only">
+        <label for="forwardOnly">
+          Only
+          <span class="accesskey-first">future</span>
+          dates:
+        </label>
+        <input
+          id="forwardOnly"
+          type="checkbox"
+          class="input task-list-item-checkbox tasks-modal-checkbox"
+          accesskey="f" />
       </div>
-    </form>
-  </div>
+    </section>
+    <hr />
+    <section class="tasks-modal-dependencies-section">
+      <label for="blockedBy" class="accesskey-first">Before this</label>
+      <span class="input">
+        <input
+          accesskey="b"
+          id="blockedBy"
+          class="input"
+          type="text"
+          placeholder="Search for tasks that the task being edited depends on..." />
+        <iframe
+          style="
+            display: block;
+            position: absolute;
+            top: 0;
+            left: 0;
+            width: 100%;
+            height: 100%;
+            overflow: hidden;
+            border: 0;
+            opacity: 0;
+            pointer-events: none;
+            z-index: -1;
+          "
+          aria-hidden="true"
+          tabindex="-1"
+          src="about:blank"></iframe>
+      </span>
+      <div class="task-dependencies-container results-dependency"></div>
+      <label for="blocking">
+        Aft
+        <span class="accesskey">e</span>
+        r this
+      </label>
+      <span class="input">
+        <input
+          accesskey="e"
+          id="blocking"
+          class="input"
+          type="text"
+          placeholder="Search for tasks that depend on this task being done..." />
+        <iframe
+          style="
+            display: block;
+            position: absolute;
+            top: 0;
+            left: 0;
+            width: 100%;
+            height: 100%;
+            overflow: hidden;
+            border: 0;
+            opacity: 0;
+            pointer-events: none;
+            z-index: -1;
+          "
+          aria-hidden="true"
+          tabindex="-1"
+          src="about:blank"></iframe>
+      </span>
+      <div class="task-dependencies-container results-dependency"></div>
+    </section>
+    <hr />
+    <section class="tasks-modal-dates-section">
+      <label for="status">
```

Comment:
1. I don't think this is new, but WebStorm is saying:

![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/4840096/fd711cdd-0c04-4481-96bf-60d10a01e235)

2. It also prompted me to wonder why Status is still inside a Dates section...

Reply:
> 1. I don't think this is new, but WebStorm is saying:
> 
> ![image](https://private-user-images.githubusercontent.com/4840096/328329207-fd711cdd-0c04-4481-96bf-60d10a01e235.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwNjg5NTAsIm5iZiI6MTcxNTA2ODY1MCwicGF0aCI6Ii80ODQwMDk2LzMyODMyOTIwNy1mZDcxMWNkZC0wYzA0LTQ0ODEtOTZiZi02MGQxMGEwMWUyMzUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDdUMDc1NzMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2M2NWE5OTMzM2I2OTZhYTFlMTBhMTEyYjBmNzhhMGFlMjNjOGVhOGZkYTZmZmJhODFhYzBiMzNmYjFiNzFiYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.liNaLijNidxjNCioCG4uMSmLAmLgxYcNwfM4KaOnxPI)

Yup, this is not new, I want to fix this. While the fix is quite simple, I wish first to dig into the impact it would have.

> 2. It also prompted me to wonder why Status is still inside a Dates section...

This is a good question. This is UI, so let's talk about this in UI terms. The status and done, cancelled, created date are all in one section since they are connected - setting the status sets one of the dates, so it makes sense to have them grouped. So from a UI perspective this does makes sense. I guess you are concerned about the fact that the `<section class="tasks-modal-dates"` has status inside, which makes little sense from a programmer's perspective. But I have an idea, how to solve this =)

Actually, it will be the same thing with the recurrence - it is connected with the due, scheduled and start date logically. So I would put recurrence under the same section with due, scheduled and start date, instead of separating them with a line as it is done now.